mypy/checker.py

```
@@ -2174,7 +2174,10 @@ def try_infer_partial_generic_type_from_assignment(self,
             rvalue_type = get_proper_type(rvalue_type)
             if isinstance(rvalue_type, Instance):
                 if rvalue_type.type == typ.type:
-                    var.type = rvalue_type
+                    if is_valid_inferred_type(rvalue_type):
+                        var.type = rvalue_type
+                    else:
+                        var.type = self.inference_error_fallback_type(rvalue_type)
```

What if one just assigns the variable twice (without reading it), would it silence the error on first assignment? Not that it is too bad, but still may be surprising.

---

Yeah, a good catch! This behavior is inconsistent. I'll try to fix it.